staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -107,7 +108,14 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				buf = buf[:runtime.Stack(buf, false)]
 				err = fmt.Sprintf("%v\n%s", err, buf)
 			}
-			resultCh <- err
+			// resultCh will have no reader if the enclosing function
+			// has already exited.
+			select {
+			case resultCh <- err:
+				return
+			default:
+				utilruntime.HandleError(err)
```

Comment:
only log if err is non-nil, right?

Reply:
Yeah, in fact, it doesn't return an error at all, which makes this updated code super ugly.